homeassistant/components/splunk/__init__.py

```diff
@@ -75,16 +64,50 @@ def setup(hass, config):
     name = conf.get(CONF_NAME)
     entity_filter = conf[CONF_FILTER]
 
-    if use_ssl:
-        uri_scheme = "https://"
-    else:
-        uri_scheme = "http://"
-
-    event_collector = f"{uri_scheme}{host}:{port}/services/collector/event"
-    headers = {AUTHORIZATION: f"Splunk {token}"}
+    event_collector = SplunkSender(
+        host=host,
+        port=port,
+        token=token,
+        hostname=name,
+        protocol=["http", "https"][use_ssl],
+        verify=(use_ssl and verify_ssl),
+        api_url=SPLUNK_ENDPOINT,
+        retry_count=1,
+    )
+
+    payload = {
+        "time": time.time(),
+        "host": name,
+        "event": {
+            "domain": DOMAIN,
+            "meta": "Splunk integration has started",
+        },
+    }
 
-    def splunk_event_listener(event):
+    try:
+        event_collector._send_to_splunk(  # pylint: disable=protected-access
```

Comment:
We need to schedule this on the executor too.

Reply:
I am pushing a change that replaces this to be fully async now.

Reply:
Great!